lib/c-api/src/wasm_c_api/engine/config/jsc.rs

```diff
@@ -0,0 +1,38 @@
+use crate::{
+    error::update_last_error,
+    wasm_c_api::engine::{wasm_config_t, wasm_engine_t, wasmer_engine_t},
+};
+
+use super::wasmer_engine_config_t;
+
+/// Configuration specific for the `jsc` engine.
+///
+/// This is a Wasmer-specific type with Wasmer-specific functions for
+/// manipulating it.
+///
+/// cbindgen:ignore
+#[repr(C)]
+#[derive(Debug, Default)]
+pub struct wasmer_jsc_engine_config_t;
+
+/// Create a new  [`wasm_engine_t`] backed by a `jsc` engine.
+pub fn wasm_jsc_engine_new_with_config(config: wasm_config_t) -> Option<Box<wasm_engine_t>> {
+    if !matches!(config.engine, wasmer_engine_t::JSC) || !config.engine_config.is_jsc() {
+        update_last_error("Cannot create a new `sys` engine with a non-sys-specific config!");
+        return None;
+    }
+
+    Some(Box::new(wasm_engine_t {
+        inner: wasmer_api::jsc::Engine::default().into(),
+    }))
+}
```

Comment:
This function is public, therefore why are we checkign the engine if what we believe it should be? I'd expect to receive a `wasm_jsc_config_t` instead of a generic `wasm_config_t`

Reply:
Because this would mean introducing new `wasm(er)_config_js_new`. Currently we only have `wasm_config_new`, which returns a `wasm_config_t`. With the current API, the check is justified by the fact that code like 
```
let mut config = wasm_config_new();
wasm_config_set_engine(&mut config, V8);
wasm_config_sys_set_target(&mut config, my_target); 
``` 
does not produce type errors. 

Do you suggest to introduce the `wasm(er)_config_<backend>_new` functions? This would also mean duplicating the `features` field that we currently have in the `wasm_config_t` type.

Reply:
Then why is `wasm_jsc_engine_new_with_config` public and how is expected to be used?

I'd expect the users just call `wasm_engine_new_with_config` and never use the other function

Reply:
Ah, that explains it. All these engine-specific config functions should be `pub(crate)` and used only in the top-level `wasm_engine_new_with_config` and not exposed to users.